### ISSUE TITLE
Some documentation ?

### ISSUE THREAD

**Issue by SwiTool** [awslabs/aws-sdk-cpp#59]
Hey,

it looks like you have done a greaaaat job !
But I need some documentation. I'm using EC2 and I don't really know what to write to manipulate instances..

Could you clarify the thing please ?

Thank you

**Comment by prestomation** [comment]
Hi SwiTool,

This repo is for the C++ SDK. This helps you call AWS from your C++ application. You still need to familiarize yourself with the actual services to understand what APIs you want to call and when.

You can find the Amazon EC2 documentation here: https://aws.amazon.com/documentation/ec2/. 

Once you know what you want to do with the EC2 API, you can use the AWS SDK for C++  to call Amazon EC2 from your C++ application.

**Comment by SwiTool** [comment]
Hi prestomation.

I need to describe, run and stop instances.
I've already done some instances manipulation with CLI Tools on Linux, but it required too much parsing and stuff not really accurate.

I'm switching to C++ SDK to avoid this inaccuracy, but by coming here, I don't see any doc, and by trying to make an Aws::EC2::EC2Client I see that I need Aws::Auth::AWSCredentials (thanks Visual Studio for auto-completion and function-helpers...), and so on...

So basically I need to know the usage of each method.

Is there any wiki for this SDK ?

Thank you

**Comment by prestomation** [comment]
Hi SwiTool,

The C++ SDK is best suited for high-performance applications where you are trying the squeeze the most performance out of your hardware OR for when you have an existing code base in C++ that you want to integrate with AWS.

If you are just getting started with AWS and don't require the high-performance nature of C++, I would recommend using an AWS SDK in a managed language in Python or Ruby or whatever you are comfortable with, you will probably find it easier to use and much better documented. This SDK is new and so you will find much less sample code and documentation as compared to the other SDKs.

That said, you should find the README useful. It covers the features of the SDK
: https://github.com/awslabs/aws-sdk-cpp/blob/master/README.md 
For basic credential management, the SDK works like all the other AWS SDKs.
You may also look at the integration test packages for more sample code.

**Comment by SwiTool** [comment]
Got it.
Thank you for the complete answer. I really appreciate it.

Will do it in one of those languages that Amazon is compatible with.
